client/app/views/contacts/index.coffee

```diff
@@ -75,11 +69,19 @@ module.exports = class Contacts extends Mn.CompositeView
             @renderChildView view, index
             view._parent = @
 
+            nextModel = models.shift()
+            if nextModel
+                index++
+                setTimeout -> _render nextModel
+            else
+                @trigger 'show:collection'
+
+        model = models.shift()
+        setTimeout -> _render model
```

Comment:
I think it can be shortened to something like:

```coffee
showCollection: ->
    index = 0
    models = _.clone @_getModels()

    _render = =>
        return @trigger 'show:collection' unless models.length
        model = models.shift()
        view  = @buildChildView model
        @_updateIndices view, true, index
        @children.add view
        @renderChildView view, index
        view._parent = @
        index++
        setTimeout _render

    _render()
```